packages/statemanager/package.json

```diff
@@ -41,12 +41,32 @@
     "tsc": "../../config/cli/ts-compile.sh"
   },
   "dependencies": {
+    "@ethereumjs/block": "^4.0.0",
```

Comment:
I am not seeing on first sight (and would generally want to avoid it) why the Block library is added here as a main dependency?

Reply:
I had it here because I have the `getBlockFromRpc` method that takes care of returning pulling a block from the provider, getting the uncle headers (if needed), and then putting it in our `block` format so it's ready to go in a VM run.  We could certainly remove the dependency on `block` and just have it return an RPC format block that then put the onus on the user to call `Block.blockFromRpc` if they want to use `vm.runBlock`.  Is that your preference here?

Reply:
I think we should move this whole method over to the `Block` library, this is the part which finally overloads the `StateManager` and is misplaced, also this `Block` dependency is not fitting (a state manager should not depend on a block?).

Since we wanted to improve on Ethers compatibility anyhow: I would suggest that we directly integrate this as a new static constructor `Block.fromEthersRPCProvider(provider, blockTag, opts?: BlockOptions)` (or similar). This is likely generally useful for people. I would have a tendency to do for Tx as well the same thing.

Optimally this should be done in a separate PR and then this PR rebased on top. 😋

And regarding a monorepo-wide Ethers strategy:

I would have a tendency to take `Ethers` in into the monorepo as a relatively "standard" dependency like we have with `ethereum-cryptography`, so also aligning on the version numbers.

This would lead to the following two points:

1. I think we should directly **take "whole" Ethers in** an not only the provider. Then we are more future proof on future use cases and can e.g. easily add a static constructor `Block.fromEthers()` (so one for just the block object, also re-occuring use case). And Ethers is just not such a heavy burden as a dependency, I guess this should be ok.
2. I would have a strong tendency to **directly go for Ethers v6** to be future proof. Final release should be very very imminent.

WDYGT?

Reply:
Have made all these moves in block and tx.  Now I just need to figure out what's going on when running some blocks.  I found that 2000005 is returning an incorrect gas calculation (likely related somehow to deleting trie nodes or something but not sure).